modules/db/src/test/java/org/mule/module/db/internal/domain/connection/SimpleConnectionFactoryTestCase.java

```diff
@@ -44,6 +49,22 @@ public void failsOnConnectionError() throws Exception
 
         connectionFactory.create(dataSource);
     }
+    
+    @Test(expected = ConnectionCreationException.class)
+    public void failsOnConnectionThenLogWithoutCredentials() throws Exception
+    {
+        when(dataSource.getConnection()).thenThrow(new RuntimeException(EXCEPTION_MESSAGE));
+
+        try
+        {
+            connectionFactory.create(dataSource);
+        }
+        catch (ConnectionCreationException e)
+        {
+            assertThat(e.getMessage(), equalTo(EXPECTED_EXCEPTION_MESSAGE));
+            throw e;
```

Comment:
This does not follow the pattern used to test error conditions. If you want to test for an error condition you must expect an exception in the test only if you are not catching it. If this case, as you are catching the expected exception and asserting the message on it, then there is no need to re throw it. You must only add a fail after the method that is expected to throw the exception to avoid having the test succeeding when is not thrown.